lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1250,3 +1254,82 @@ def delete_comment(request, comment_id):
         comment_deleted.send(sender=None, user=request.user, post=cc_comment)
     else:
         raise PermissionDenied
+
+
+def _get_course_discussion_user_stats(
+    course_key: CourseKey,
+    order_by: UserOrdering = UserOrdering.BY_FLAGS,
+) -> List[Dict[str, Union[str, int]]]:
+    """
+    Get course user discussion stats.
+
+    Args:
+        course_key (CourseKey): course key for course to fetch discussion stats for.
+        order_by (UserOrdering): the ordering to use for sorting data
+
+    Returns:
+        Sorted user stat data for the specified course
+    """
+    course_stats = get_course_user_stats(course_key)
+
+    def key_func(user):
+        key = (user["threads"], user["responses"], user["replies"])
+        if order_by == UserOrdering.BY_FLAGS:
+            key = (user["active_flags"], user["inactive_flags"]) + key
+        return key
+
+    return sorted(course_stats.values(), key=key_func, reverse=True)
```

Comment:
Can we move this sorting logic in the frontend than the backend?

Reply:
The response is paginated. I think we need to sort it from the backend

Reply:
The data will be for all users in the course. So for small enough courses it might be OK, but if a course has thousands or tens of thousands of users, that will be hard. Pagination and sorting on the backend is essential here.

Reply:
Sorry, i am not able to understand why sorting from the backend is essential. can you explain a little more @xitij2000 thanks

Reply:
To sort data in the frontend, the frontend needs to have access to all the data. In this case, it could be data for thousands of users. Without all the users, the sorting cannot be performed.

Reply:
Got it. I just saw we are paginating on the backend.